watertap/core/membrane_channel_base.py

```diff
@@ -0,0 +1,904 @@
+        doc="""
+    Indicates what type of pressure change calculation will be made. To use any of the
+    ``pressure_change_type`` options to account for pressure drop, the configuration keyword
+    ``has_pressure_change`` must also be set to ``True``. Also, if a value is specified for pressure
+    change, it should be negative to represent pressure drop.
+
+    **default** - ``PressureChangeType.fixed_per_stage`` 
+
+.. csv-table::
+    :header: "Configuration Options", "Description"
+
+    "``PressureChangeType.fixed_per_stage``", "Specify an estimated value for pressure drop across the membrane feed channel"
+    "``PressureChangeType.fixed_per_unit_length``", "Specify an estimated value for pressure drop per unit length across the membrane feed channel"
+    "``PressureChangeType.calculated``", "Allow model to perform calculation of pressure drop across the membrane feed channel"
+
+""",
+    ),
+)
+
+
+class MembraneChannelMixin:
+    def _add_pressure_change(self, pressure_change_type=PressureChangeType.calculated):
+        raise NotImplementedError()
+
+    def add_total_pressure_balances(
+        self,
+        has_pressure_change=True,
+        pressure_change_type=PressureChangeType.calculated,
+        custom_term=None,
+    ):
+        super().add_total_pressure_balances(
+            has_pressure_change=has_pressure_change, custom_term=custom_term
+        )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.length_domain,
+            doc="Pressure at interface",
+        )
+        def eq_equal_pressure_interface(b, t, x):
+            return b.properties_interface[t, x].pressure == b.properties[t, x].pressure
+
+        if has_pressure_change:
+            self._add_pressure_change(pressure_change_type=pressure_change_type)
+
+        if pressure_change_type == PressureChangeType.calculated:
+            self._add_calculated_pressure_change()
+
+    def add_isothermal_conditions(self):
+
+        # ==========================================================================
+        # Bulk and interface connections on the feed-side
+        # TEMPERATURE
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.length_domain,
+            doc="Temperature at interface",
+        )
+        def eq_equal_temp_interface(b, t, x):
+            return (
+                b.properties[t, x].temperature
+                == b.properties_interface[t, x].temperature
+            )
+
+    def add_volumetric_flowrate_balance(self):
+        # VOLUMETRIC FLOWRATE
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.length_domain,
+            doc="Volumetric flow at interface of inlet",
+        )
+        def eq_equal_flow_vol_interface(b, t, x):
+            return (
+                b.properties_interface[t, x].flow_vol_phase["Liq"]
+                == b.properties[t, x].flow_vol_phase["Liq"]
+            )
```

Comment:
We might want to change this wording. We aren't necessarily doing a volumetric flowrate balance, we are just providing extensive flow information to the interface. Maybe "add_extensive_flow_to_interface" would be best.

Reply:
Done in [4300d89](https://github.com/watertap-org/watertap/pull/679/commits/4300d89a88728121b1e43b98bbcfedf862b6b384)